scripts/menu/server/sv_menu.lua

```diff
@@ -0,0 +1,405 @@
+--Check Environment
+if GetConvar('txAdmin-serverMode', 'false') ~= 'true' then
+  return
+end
+local apiHost = GetConvar("txAdmin-apiHost", "invalid")
+local pipeToken = GetConvar("txAdmin-pipeToken", "invalid")
+if apiHost == "invalid" or pipeToken == "invalid" then
+  logError('API Host or Pipe Token ConVars not found. Do not start this resource if not using txAdmin.')
+  return
+end
+
+--Erasing the token convar for security reasons
+if (GetConvar('TXADMIN_MENU_DEBUG', 'false') ~= 'true') then
+  SetConvar("txAdmin-pipeToken", "removed")
+end
+
+
+-- Vars
+local adminPermissions = {}
+
+
+--- Determine if a source has a given permission
+---@param source number
+---@param permission string
+---@return boolean
+local function PlayerHasTxPermission(source, permission)
+  local allow = false
+  local perms = adminPermissions[source]
+  if perms then
+    for _, perm in pairs(perms) do
+      if perm == 'all_permissions' or permission == perm then
+        allow = true
+        break
+      end
+    end
+  end
+  debugPrint(string.format("permission check (src=^3%d^0, perm=^4%s^0, result=%s^0)",
+    source, permission, (allow and '^2true' or '^1false')))
+  return allow
+end
+
+AddEventHandler('txAdmin:events:adminsUpdated', function(onlineAdminIDs)
+  debugPrint('^3Admins changed. Online admins: ' .. json.encode(onlineAdminIDs) .. "^0")
+  
+  -- Collect old and new admin IDs
+  local refreshAdminIds = {}
+  for id, _ in pairs(adminPermissions) do
+    refreshAdminIds[id] = id
+  end
+  for newId, _ in pairs(adminPermissions) do
+    refreshAdminIds[newId] = newId
+  end
+  
+  -- Resetting all admin permissions
+  adminPermissions = {}
+  
+  -- Informing clients that they need to reauth
+  for id, _ in pairs(refreshAdminIds) do
+    TriggerClientEvent('txAdmin:menu:reAuth', id)
+  end
+end)
+
+-- 
+-- [[ WebPipe Proxy ]]
+--
+RegisterNetEvent('txAdmin:WebPipe')
+AddEventHandler('txAdmin:WebPipe', function(callbackId, method, path, headers, body)
+  local s = source
+
+  -- Adding auth information
+  if path == '/auth/nui' then
+    headers['X-TxAdmin-Token'] = pipeToken
+    headers['X-TxAdmin-Identifiers'] = table.concat(GetPlayerIdentifiers(s), ', ')
+  else
+    headers['X-TxAdmin-Token'] = 'not_required' -- so it's easy to detect webpipes
+  end
+  
+  local url = "http://" .. apiHost .. path:gsub("//", "/")
+  debugPrint(string.format("^3WebPipe[^5%d^0:^1%d^3]^0 ^4>>^0 ^6%s^0", s, callbackId, url))
+  debugPrint(string.format("^3WebPipe[^5%d^0:^1%d^3]^0 ^4>>^0 ^6Headers: %s^0", s, callbackId, json.encode(headers)))
+
+  PerformHttpRequest(url, function(httpCode, data, resultHeaders)
```

Comment:
Potential for some nasty abuse from event handler submission here. I'd stick to only allowing `/auth/nui` for players that aren't in the `adminPermissions` table, and caching a failure or attempt to call it way too often, and performing a no-op otherwise.